sum_types_generator/lib/src/templates.dart

```diff
@@ -141,6 +141,33 @@ String function({
         ";",
     ].join("\n");
 
+String expressionFunction({
+  required String type,
+  required String name,
+  required Iterable<String> body,
+  bool isStatic = false,
+  Iterable<TypeParamSpec> typeParams = const [],
+  Iterable<String> posParams = const [],
+  Iterable<String> namedParams = const [],
+}) =>
+    [
+      if (isStatic) "static ",
+      type,
+      " ",
+      name,
+      parametrize(typeParams),
+      "(",
+      ...posParams.map(appendComma),
+      if (namedParams.isNotEmpty) ...[
+        "{",
+        ...namedParams.map(appendComma),
+        "}",
+      ],
+      ") =>",
+      body.join("\n"),
+      "  ",
```

Comment:
Hopefully, the last thing. Sorry, didn't catch all in one go.

This space is not needed, but instead a semicolon should (probably) become a part of this template.

An expression function (not a closure), either top-level or a class member, always ends with a semicolon:

```dart
int f() => 1;
```

I'd argue that the semicolon here is a part of the "outer" expression-function syntax, rather than a part of the expression (i.e. I'd consider only `1` to be an expression in the example above). Semicolons typically separate statements, and they are typically not used with expressions.

If you add the trailing semicolon to this template, please make sure the generator doesn't end up generating two semicolons, because another one is inserted somewhere around the call site of this template (it will probably be a compile-time error, so you shouldn't miss it).